juju/juju
# Fix/caas relation datebag inconsistency

## Description of change

Fix: statefulset pod units are been removed then cause relation data bag lost;

## QA steps

- deploy mediawiki
- kill pod;

```bash
$ kubectl delete pod/mariadb-k8s-0 -n $NS
```
- monitor juju status to confirm, the unit number is not changed;

- add/remove units;

```bash
$ juju add-unit mariadb-k8s -n 1
$ juju remove-unit mariadb-k8s --num-units=2
```

## Documentation changes

None

## Bug reference

https://bugs.launchpad.net/juju/+bug/1830745
